src/DataTypes/DataTypeObject.cpp

```diff
@@ -336,16 +323,16 @@ std::unique_ptr<ISerialization::SubstreamData> DataTypeObject::getDynamicSubcolu
     /// we should return JSON column with data {"c" : {"d" : 10, "e" : Hello}, "f" : [1, 2, 3]}
     if (auto sub_object_subcolumn = tryGetSubObjectSubcolumn(subcolumn_name))
     {
-        const String & prefix = *sub_object_subcolumn;
+        const String & prefix = *sub_object_subcolumn + ".";
         /// Collect new typed paths.
         std::unordered_map<String, DataTypePtr> typed_sub_paths;
         /// Collect serializations for typed paths. They will be needed for sub-object subcolumn deserialization.
         std::unordered_map<String, SerializationPtr> typed_paths_serializations;
         for (const auto & [path, type] : typed_paths)
         {
-            if (path.starts_with(prefix) && path.size() != prefix.size())
+            if (path.starts_with(prefix))
```

Comment:
Do we validate `path` anywhere? Is it possible to pass something like `a.b.` or `a.b..`?

Reply:
Whay do you mean by validating?
Paths `a.b.` or `a.b..` are valid, it means that there are objects with empty keys `{"a" : {"b" : {"" : {"" : ...}}}}`